Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1566,4 +1566,80 @@ function Set-ToolsRepo
             Write-Host "Successfully set tools-repo on all hosts for datastore $ds_name"
         }
     }
-}
\ No newline at end of file
+}
+
+<#
+.Synopsis
+    Set vSAN compression and deduplication on a cluster or clusters. If deduplication is enabled then compression is required.
+    The default cluster configuration is deduplication and compression but the customer can change that.
+    Choosing neither compression nor deduplication will disable both.
+    This requires action on every physical disk and will take time to complete.
+
+.EXAMPLE
+    Set-vSANCompressDedupe -compression -cluster "cluster-1", "cluster-2"
+
+    Set-vSANCompressDedupe -deduplication -cluster "cluster-1", "cluster-2"
+
+    Set-vSANCompressDedupe -cluster "cluster-1", "cluster-2"
+
+    Set-vSANCompressDedupe -cluster "all"
+#>
+
+function Set-vSANCompressDedupe
+{
+    [AVSAttribute(120, UpdatesSDDC = $true)]
+    param(
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Enable compression and deduplication.")]
+        [bool]$deduplication = $false,
```

Comment:
These parameters become UI elements, let's follow PowerShell guidelines and start the names with Uppercase